config/connect-avro-distributed.properties

```diff
@@ -17,26 +17,46 @@ key.converter.schema.registry.url=http://localhost:8081
 value.converter=io.confluent.connect.avro.AvroConverter
 value.converter.schema.registry.url=http://localhost:8081
 
-# The internal converter used for offsets and config data is configurable and must be specified,
-# but most users will always want to use the built-in default. Offset and config data is never
-# visible outside of Connect in this format.
+# Internal Storage Topics.
+#
+# Kafka Connect distributed workers store the connector and task configurations, connector offsets,
+# and connector statuses in three internal topics. These topics MUST be compacted.
+# When the Kafka Connect distributed worker starts, it will check for these topics and attempt to create them
+# as compacted topics if they don't yet exist, using the topic name, replication factor, and number of partitions
+# as specified in these properties, and other topic-specific settings inherited from your brokers'
+# auto-creation settings. If you need more control over these other topic-specific settings, you may want to
+# manually create these topics before starting Kafka Connect distributed workers.
+#
+# The following properties set the names of these three internal topics for storing configs, offsets, and status.
+config.storage.topic=connect-configs
+offset.storage.topic=connect-offsets
+status.storage.topic=connect-statuses
+
+# The following properties set the replication factor for the three internal topics, defaulting to 3 for each
+# and therefore requiring a minimum of 3 brokers in the cluster. Since we want the examples to run with
+# only a single broker, we set the replication factor here to just 1. That's okay for the examples, but
+# ALWAYS use a replication factor of AT LEAST 3 for production environments to reduce the risk of 
+# losing connector offsets, configurations, and status.
+config.storage.replication.factor=1
+offset.storage.replication.factor=1
+status.storage.replication.factor=1
+
+# The config storage topic must have a single partition, and this cannot be changed via properties. 
+# Offsets for all connectors and tasks are written quite frequently and therefore the offset topic
+# should be highly partitioned; by default it is created with 25 partitions, but adjust accordingly
+# with the number of connector tasks deployed to a distributed worker cluster. Kafka Connect records
+# the status less frequently, and so by default the topic is created with 5 partitions.
+#offset.storage.partitions=25
+#status.storage.partitions=5
+
+# The offsets, status, and configurations are written to the topics using converters specified through
+# the following required properties. Most users will always want to use the JSON converter without schemas. 
+# Offset and config data is never visible outside of Connect in this format.
 internal.key.converter=org.apache.kafka.connect.json.JsonConverter
 internal.value.converter=org.apache.kafka.connect.json.JsonConverter
 internal.key.converter.schemas.enable=false
 internal.value.converter.schemas.enable=false
```

Comment:
I've moved the internal converter configs _after_ the rest of the internal topic configs for a few reasons:
* It flows a bit better.
* We recommend most users use these values as-is, so they're lower priority and can be lower in the file.
* Separating the _internal_ converter from the _external_ is good, since I've seen people get confused and update the both sets of converters when they just want to update the normal key and value converters.